frontend/src/lib/k8s/cluster.ts

```diff
@@ -146,7 +146,15 @@ makeKubeObject<T extends (KubeObjectInterface | KubeEvent)>(objectName: string):
       [U[] | null, ApiError | null, (items: U[]) => void, (err: ApiError | null) => void] {
       const [objList, setObjList] = React.useState<U[] | null>(null);
       const [error, setError] = useErrorState(setObjList);
-      this.useApiList(setObjList, setError);
+
+      function setList(items: U[] | null) {
+        setObjList(items);
+        if (items !== null) {
```

Comment:
Just a nitpick: As null is a falsy so this can be reduced down to 
```suggestion
        if (!items) {
```

Reply:
It's different. I really want to check whether it's null.